src/duration.js

```diff
@@ -433,6 +458,53 @@ export default class Duration {
     return s;
   }
 
+  /**
+   * Returns an ISO 8601-compliant string representation of this Duration, formatted as a time of day.
+   * Note that this will return null if the duration is invalid, negative, or equal to or greater than 24 hours.
+   * @see https://en.wikipedia.org/wiki/ISO_8601#Times
+   * @param {Object} opts - options
+   * @param {boolean} [opts.suppressMilliseconds=false] - exclude milliseconds from the format if they're 0
+   * @param {boolean} [opts.suppressSeconds=false] - exclude seconds from the format if they're 0
+   * @param {boolean} [opts.includePrefix=false] - include the `T` prefix
+   * @param {string} [opts.format='extended'] - choose between the basic and extended format
+   * @example Duration.fromObject({ hours: 11 }).toISOTime() //=> '11:00:00.000'
+   * @example Duration.fromObject({ hours: 11 }).toISOTime({ suppressMilliseconds: true }) //=> '11:00:00'
+   * @example Duration.fromObject({ hours: 11 }).toISOTime({ suppressSeconds: true }) //=> '11:00'
+   * @example Duration.fromObject({ hours: 11 }).toISOTime({ includePrefix: true }) //=> 'T11:00:00.000'
+   * @example Duration.fromObject({ hours: 11 }).toISOTime({ format: 'basic' }) //=> '110000.000'
+   * @return {string}
+   */
+  toISOTime(opts = {}) {
+    if (!this.isValid) return null;
+
+    const millis = this.toMillis();
+    if (millis < 0 || millis >= 86400000) return null;
+
+    opts = Object.assign({
+      suppressMilliseconds = false,
+      suppressSeconds = false,
+      includePrefix = false,
+      format = "extended"
+    }, opts);
+
+    const value = this.shiftTo("hours", "minutes", "seconds", "milliseconds");
+
+    let fmt = format === "basic" ? "hhmm" : "hh:mm";
+
+    if (!opts.suppressSeconds || value.seconds !== 0 || value.milliseconds !== 0) {
+      fmt += format === "basic" ? "ss" : ":ss";
+      if (!opts.suppressMilliseconds || value.milliseconds !== 0) {
+        fmt += ".SSS";
+      }
+    }
+
+    if (opts.includePrefix) {
+      fmt = "T" + fmt;
+    }
+
+    return value.toFormat(fmt);
+  }
+
```

Comment:
This method is designed to behave just like the existing `toISOTime()` in the `DateTime` class - and that one has been extended to support adding the prefix, but is otherwise unchanged.

In my first implementation, the default was to automatically choose whether to add the prefix, depending on the format. However, I've since decided to drop that, and instead make it an explicit option, defaulting to false.

Technically, the 2019 version of the spec does require the prefix to be present for the basic format `hhmm`, but the previous version didn't, and changing it now would therefore have been a breaking change.

Also, since the user already has to specify options to get the basic format, they might as well also specify if they want the prefix...